backends/graphics/openglsdl/openglsdl-graphics.cpp

```diff
@@ -348,29 +348,30 @@ bool OpenGLSdlGraphicsManager::loadVideoMode(uint requestedWidth, uint requested
 	Common::Rect desktopRes = _window->getDesktopResolution();
 
 #if SDL_VERSION_ATLEAST(2, 0, 0)
-	SDL_Window *window = _window->getSDLWindow();
-	bool isMaximized = window ? (SDL_GetWindowFlags(window) & SDL_WINDOW_MAXIMIZED) : false;
```

Comment:
I'm pretty sure we have to keep the `isMaximized` check intact, since this is not covered by `_wantsFullScreen`. At least on Win32 and Linux, fullscreen and maximized are not the same like on macOS, but two different modes.

Reply:
Now that I think of it, maybe it's better to fix what causes my workaround to be necessary in the first place 😂